File: frontend/lib/features/invitations/models/invitation_dto.dart
class InvitationDto {
  final String id;
  final String doctorId;
  final String doctorName;
  final int? patientDni;
  final String? patientEmail;
  final String status;
  final String? createdAt;
  final String? expiresAt;

  InvitationDto({
    required this.id,
    required this.doctorId,
    required this.doctorName,
    this.patientDni,
    this.patientEmail,
    required this.status,
    this.createdAt,
    this.expiresAt,
  });

  factory InvitationDto.fromJson(Map<String, dynamic> json) {
    return InvitationDto(
      id: json['id']?.toString() ?? '',
      doctorId: json['doctorId']?.toString() ?? '',
      doctorName: json['doctorName']?.toString() ?? '',
      patientDni: json['patientDni'] != null
          ? int.tryParse(json['patientDni'].toString())
          : null,
      patientEmail: json['patientEmail']?.toString(),
      status: json['status']?.toString() ?? 'PENDING',
      createdAt: json['createdAt']?.toString(),
      expiresAt: json['expiresAt']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'doctorId': doctorId,
      'doctorName': doctorName,
      'patientDni': patientDni,
      'patientEmail': patientEmail,
      'status': status,
      'createdAt': createdAt,
      'expiresAt': expiresAt,
    };
  }
}

File: backend/backend-dialysis-record/src/main/java/com/agustin/backend_dialysis_record/dto/InvitationDto.java
package com.agustin.backend_dialysis_record.dto;

import com.agustin.backend_dialysis_record.model.InvitationStatus;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
public class InvitationDto {
    private UUID id;
    private UUID doctorId;
    private String doctorName;
    private Integer patientDni;
    private String patientEmail;
    private InvitationStatus status;
    private LocalDateTime createdAt;
    private LocalDateTime expiresAt;
}
